### ISSUE TITLE
Null Pointer Exception java.lang.String com.stripe.stripeterminal.Reader.getSerialNumber()

### ISSUE THREAD

**Issue by legomanaustralia** [stripe/stripe-terminal-android#36]
We've noticed the following error on occasion, it looks like its related to having a bad reader connection. 

```
java.lang.NullPointerException: Attempt to invoke virtual method 'java.lang.String com.stripe.stripeterminal.Reader.getSerialNumber()' on a null object reference (Most recent call first)
at com.stripe.stripeterminal.TerminalListenerProxy.onUnexpectedReaderDisconnect (TerminalListenerProxy.java:36)
at com.stripe.stripeterminal.TerminalStatusManager.unexpectedDisconnect (TerminalStatusManager.java:71)
at com.stripe.stripeterminal.BbposBluetoothAdapter.onDisconnectReader (BbposBluetoothAdapter.java:212)
at com.stripe.stripeterminal.BbposDeviceControllerListener.onBTDisconnected (BbposDeviceControllerListener.java:81)
at com.bbposstripe.bbdevice.BBDeviceController$111.run (SourceFile:2925)
```

**Comment by SharathKaribasappaGB** [comment]
I have also observed this issue occasionally. Same stack traces as above.

**Comment by danwaters-stripe** [comment]
Thanks for the report. This is due to an odd circumstance where the device disconnects before it actually finishes connecting. I'll be sure to fix the NPE in the next release, and will continue to investigate the cause of these failed connections.

**Comment by danwaters-stripe** [comment]
This has been fixed in 1.0.0-b5.
